pkg/container-collection/options.go

```diff
@@ -819,6 +819,9 @@ func WithTracerCollection(tc TracerCollection) ContainerCollectionOption {
 					container.Runtime.ContainerID, err)
 				return false
 			}
+			if container.mntNsFd != 0 {
+				log.Warnf("WithTracerCollection: mntns reference already set for container %s", container.Runtime.ContainerID)
+			}
 			container.mntNsFd = mntNsFd
```

Comment:
Is the reassignment of `container.mntNsFd` enough to close (GC) the previous fd?

Reply:
No, that would not close the previous fd if a previous fd exists.

If the previous `mntNsFd` is not zero, that's a bug, it is not supposed to happen. The bug could either be that there is a previous fd, or that due to some memory corruption, `mntNsFd` is assigned to a non-zero number that is not an open fd, or it could be another fd unrelated to mount namespaces that is used by other code. In that latter case, I thought it would be less harmful to potentially leak a fd rather than potentially close an unrelated fd.

What do you think?

Reply:
Another case is if we introduce a regression and somehow we start calling the enrichers twice again. But it is supposed to be detected by the tests you added. I just tried and it does.

> In that latter case, I thought it would be less harmful to potentially leak a fd rather than potentially close an unrelated fd.
>
Yeah, but I think it safer to not close fds that we didn't open.